### DISCUSSION TITLE
ScrollBar control cannot slide

### DISCUSSION THREAD

**Discussion by pyking-wang** [AvaloniaUI/Avalonia#12263] [Q&A] [answered]
It is also normal in the previewer, but cannot slide after compilation.
`<ScrollBar Visibility="Visible" Orientation="Horizontal" Scroll="ScrollHandler1" Grid.Column="1"/>`
![GIF 2023-7-20 10-55-42](https://github.com/AvaloniaUI/Avalonia/assets/76192669/6d40e63c-cbb8-42b2-bd75-88de3e2d90e8)
The slider has disappeared.
![image](https://github.com/AvaloniaUI/Avalonia/assets/76192669/54d407d3-1553-42ec-8783-0d61aa23c12a)

**Comment by timunie** [comment]
looks like the datacontext maybe wrong? Please check with DevTools. If the DataContext is correct, please file a minimal sample to reproduce.

**Reply by pyking-wang** [reply]
Hello, I conducted a test and I seem to have found the problem. It is normal under window, but this issue may occur under usercontrol (the preview is normal, but the bar disappears during runtime).
This problem can be reproduced using the simplest demo.
I think this seems like a bug.

**Reply by timunie** [reply]
please provide me the sample. I can convert this into an issue if I can judge. Thx.

**Reply by pyking-wang** [reply]
1. TTSChoice.axaml
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="clientbyavalonia.Views.TTSChoice">
	<Grid ColumnDefinitions="*,Auto" Height="20" Width="300">
		<ScrollBar Visibility="Visible" Orientation="Horizontal" Scroll="ScrollHandler"></ScrollBar>
		<TextBlock Name="valueText" Grid.Column="1"/>
	</Grid>
</UserControl>
```
2. TTSChoice.axaml.cs
```
using Avalonia;
using Avalonia.Controls;
using Avalonia.Controls.Primitives;
using Avalonia.Markup.Xaml;

namespace clientbyavalonia.Views;

public partial class TTSSChoice : UserControl
{
    public TTSChoice()
    {
        InitializeComponent();
    }
    public void ScrollHandler(object source, ScrollEventArgs args)
    {
        valueText.Text = args.NewValue.ToString();
    }
}
```
3.MainWindow.axaml
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="clientbyavalonia.Views.MainWindow"
        Title="test"
	xmlns:local="using:clientbyavalonia.Views">
	<Button Content="ClickMe">		
		<Button.Flyout>
			<Flyout Placement="Top">
				<local:TTSSChoice/>
			</Flyout>
		</Button.Flyout>
	</Button>
</Window>
```
**PS: Invalidate on page float**

**Reply by timunie** [reply]
hm could indeed be an issue. I'll open one.

**Reply by pyking-wang** [reply, marked as answer]
@timunie Hello, I noticed that this issue seems to have been fixed. I am using the latest night building. I will close this discussion and once you confirm, you can update the status of #12275.

**Reply by timunie** [reply]
thx 🙏

**Comment by pyking-wang** [comment]
Follow #12275
